public/controllers/register-agent/components/command-output/command-output.tsx

```diff
@@ -3,35 +3,68 @@ import {
   EuiCopy,
   EuiIcon,
   EuiSpacer,
+  EuiSwitch,
+  EuiSwitchEvent,
   EuiText,
 } from '@elastic/eui';
-import React, { Fragment, useState } from 'react';
+import React, { Fragment, useEffect, useState } from 'react';
 
 interface ICommandSectionProps {
   commandText: string;
   showCommand: boolean;
   onCopy: () => void;
   os?: 'WINDOWS' | string;
+  password?: string;
 }
 
 export default function CommandOutput(props: ICommandSectionProps) {
-  const { commandText, showCommand, onCopy, os } = props;
+  const { commandText, showCommand, onCopy, os, password } = props;
   const getHighlightCodeLanguage = (os: 'WINDOWS' | string) => {
     if (os.toLowerCase() === 'windows') {
       return 'powershell';
     } else {
       return 'bash';
     }
   };
-
-  const [language, setLanguage] = useState(getHighlightCodeLanguage(os || ''));
+  const [havePassword, setHavePassword] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
 
   const onHandleCopy = (command: any) => {
     onCopy && onCopy();
-    return command
+    return command; // the return is needed to avoid a bug in EuiCopy
   };
 
-  const [commandToCopy, setCommandToCopy] = useState(commandText);
+  const [commandToShow, setCommandToShow] = useState(commandText);
+
+  useEffect(() => {
+    if (password) {
+      setHavePassword(true);
+      osdfucatePassword(password);
+    } else {
+      setHavePassword(false);
+      setCommandToShow(commandText);
+    }
+  }, [password, commandText, showPassword])
+
+  const osdfucatePassword = (password: string) => {
+    if(!password) return;
+    if(!commandText) return;
+
+    if(showPassword){
+      setCommandToShow(commandText);
+    }else{
+    // search password in commandText and replace with * for every character
+      const findPassword = commandText.search(password);
+      if (findPassword > -1) {
+        const command = commandText;
+        setCommandToShow(command.replace(password, '*'.repeat(password.length)));
```

Comment:
Maybe to avoid unexpected replacements, the replacement can be improved as follows:
command.replace(/WAZUH_REGISTRATION_PASSWORD='([^']+)'/,`WAZUH_REGISTRATION_PASSWORD='${'*'.repeat(password.length)}'`)